### PR TITLE
Add more content to cni README.md

### PR DESCRIPTION
**Pull request by costinm**
I've been chasing a bug and looked a bit more at the code - added my notes to the README in case others need to do the same. It's a draft, please let me know if anything is too low level.

**Please provide a description of this PR:**

### CODE REVIEW

File: cni/README.md [hunk 1]
```diff
@@ -82,44 +85,91 @@ The following example grabs the last 10 `kubelet` logs containing the string "cm
...
+- plugin config "exclude namespaces" applies first
+- ambient is enabled if: 
+  - namespace label "istio.io/dataplane-mode" == "ambient" is required (may change for 'on-by-default' mode)
+  - "sidecar.istio.io/status" annotation is not present on the pod (created by injection of sidecar)
+  - "ambient.istio.io/redirection" is not "disabled"
+- sidecar interception is enabled if:
+  - "istio-init" container is not present in the pod.
+  - istio-proxy container exists and 
+    - does not have DISABLE_ENVOY environment variable (which triggers proxyless mode)
+    - has a istio-proxy container, with first 2 args "proxy" and "sidecar" - or less then 2 args, or first arg not proxy.
```
Comment (howardjohn) [inline cni/README.md:105]: This is definitely not fragile :slightly_smiling_face:

**Review by howardjohn — APPROVED** [review]

**Comment by costinm** [comment]
I just documented what I understand from the code - before I forget.
Doesn't mean I like it - or that I understand all the cases, but at least
it's
written down and others can fix what I got wrong..

My hope is that CNI vendors will handle this natively - with some cleaner
API, and maybe with a vendor-neutral fork of istio-cni
providing a polyfill. I think the danger is users installing 2 CNI plugins
- and then having the code that attempts to keep
the installer in a certain state and the iptables fight with each other.
But we'll need to clean up our mess.

**Pull request merged**